### ISSUE TITLE
Question: How do I explain capacity saturation? 

### ISSUE THREAD

**Issue by lppier** [facebook/prophet#798]
I am using capacity saturation  in my forecasting in prophet, and my colleagues have asked about whether capacity is a hard limit on the values (much like a brick-wall compressor in audio). From what I observe, I do see that the capacity saturation affects the predictions, but does not completely stop it from "breaking through" the capacity values. (I am doing airline analytics so my capacity is in the form of available planes) I want to use capacity as a saturator as I see the benefits in MAPE in many cases. The concern from my colleagues is that I am enforcing a hard limit on the forecasting. 

How do I explain the capacity in this case, maybe mathematically? Why are the forecasted values still able to "break" through? Attached is an example of what I mean.  

![image](https://user-images.githubusercontent.com/1635202/50755403-d8e6af00-1293-11e9-8521-de52ebda029d.png)

Please also do let me know if I have done something horribly wrong!

Thank you. 

TLDR - It would be great if someone could help me "justify" how the capacity does not enforce a hard limit on the forecasted values.

**Comment by bletham** [comment]
The capacity sets a hard maximum on the trend component. Mathematically, it uses the logistic growth function for the trend component: https://en.wikipedia.org/wiki/Logistic_function , so the trend will never have a value larger than L (in that page's notation), or less than 0.

If you look at the components plot (`prophet_plot_components(m, forecast)`), then you will see that the trend component saturates at and is always less than the capacity.

However, the capacity applies only to the trend, and so other components of the model can put the total prediction (`yhat`) above the capacity. With additive seasonality, the model is
```
yhat(t) = trend(t) + seasonality(t)
```
So the seasonality fluctuates around the trend, and can cross the capacity line. Similarly with multiplicative seasonality,
```
yhat(t) = trend(t)* (1 + seasonality(t))
```
and so the seasonal effects are still applied on top of the saturating trend.

Let me know if that makes sense. In this particular case, I think if you look at the components plot it should be a lot more clear.

**Comment by lppier** [comment]
Thank you @bletham , it is super clear now. I had neglected the fact that the capacity only saturates the trend.
